### PR TITLE
Makman2/reqs

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

**Comment by Makman2** [comment]
Should fix https://github.com/coala/coala-bears/issues/49

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2843/commits/4b23224569751c157c08196af7fa9ec6c3a46906

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2843/commits/dd38c19e333ebb2fad702178f32b478c5d1c75c2

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2843/commits/973c6c7974254df7b2e1974a8f5f6a7edb7d62ab

**Comment by sils** [comment]
@Makman2 coverage is failing

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2843/commits/91e3c17305c66ed97118ab08ef1fe585c1c2b25c

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2843/commits/1833807cab7924447484bd4561511c768e84e8e0 https://github.com/coala/coala/pull/2843/commits/40c08ebe0d433544749f88127501c9b8b166baa1

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -336,14 +336,33 @@ def check_prerequisites(cls):
...
+                    "Requirement for '{}' ({}) is not satisfied.".format(
+                        requirement.package, requirement.version))
+
+                if isinstance(requirement, PackageRequirement):
+                    try:
+                        # FIXME What about shell-quotation for returned
+                        # FIXME `install_command`?
+                        error_string += (
+                            '\nYou can install it using this command:\n' +
+                            ' '.join(requirement.install_command))
```
Comment (sils) [inline coalib/bears/Bear.py]: uhm all requirements should have an install command, this shouldn't be specific to the package requirement

**Comment by sils** [comment]
unack https://github.com/coala/coala/pull/2843/commits/739f7d4b6c5820f94b92399322cecdf219a1e72b

File: coalib/bears/Bear.py (hunk 1 above)
Comment (Makman2) [inline coalib/bears/Bear.py]: no only `DistributionRequirement`s have one, normal `PackageRequirement`s don't have an `install_command` out of the box^^

File: coalib/bearlib/abstractions/Linter.py [hunk 2]
```diff
@@ -185,8 +185,12 @@ def check_prerequisites(cls):
             :return:
                 True if operational, otherwise a string containing more info.
             """
+            basecheck_result = LocalBear.check_prerequisites()
+            if basecheck_result is not True:
+                return basecheck_result
+
             if shutil.which(cls.get_executable()) is None:
```
Comment (Makman2) [inline coalib/bearlib/abstractions/Linter.py]: already thought about this too. Will do that, though that interferes with https://github.com/coala/coala/issues/2073 a bit^^

File: coalib/bears/Bear.py (hunk 1 above)
Comment (Makman2) [inline coalib/bears/Bear.py]: but indeed this shouldn't be bound to a `DistributionRequirement` only, filed https://github.com/coala/coala/issues/2847

**Comment by Makman2** [comment]
Out-of-date and partially implemented.

**Pull request closed without merging**